### PR TITLE
Use of `spherical_jn` introduced

### LINKED ISSUE #12369: Start using spherical_jn from SciPy
SciPy 0.18 deprecates `sph_jn`, we should use a try/except block to try to import spherical_jn

```
$ ./bin/test sympy/external/tests/test_scipy.py -k jn_zeros
...
  File "/home/bjorn/vc/sympy/sympy/external/tests/test_scipy.py", line 26, in test_jn_zeros
    assert eq(jn_zeros(0, 4, method="scipy"),
  File "sympy/functions/special/bessel.py", line 972, in jn_zeros
    root = solver(f, root)
  File "sympy/functions/special/bessel.py", line 964, in solver
    root = newton(f, x)
  File "/home/bjorn/.local/lib/python2.7/site-packages/scipy/optimize/zeros.py", line 167, in newton
    q0 = func(*((p0,) + args))
  File "sympy/functions/special/bessel.py", line 958, in <lambda>
    f = lambda x: sph_jn(n, x)[0][-1]
  File "/home/bjorn/.local/lib/python2.7/site-packages/numpy/lib/utils.py", line 100, in newfunc
    warnings.warn(depdoc, DeprecationWarning, stacklevel=2)
DeprecationWarning: `sph_jn` is deprecated!
scipy.special.sph_jn is deprecated in scipy 0.18.0. Use scipy.special.spherical_jn instead. Note that the new function has a different signature.
```

here is the use of sph_jn:
```
$ grep --include "*.py" sph_jn -R .
./sympy/functions/special/bessel.py:      `SciPy's sph_jn <http://docs.scipy.org/doc/scipy/reference/generated/scipy.special.jn_zeros.html>`_
./sympy/functions/special/bessel.py:        from scipy.special import sph_jn
./sympy/functions/special/bessel.py:        f = lambda x: sph_jn(n, x)[0][-1]
```

### PR DESCRIPTION
**Pull request by mamakancha**
This PR fixes #12369

### CODE REVIEW

**Review by epurohit — COMMENTED** [review]
@mamakancha , Everything looks fine and should fix the issue I guess.
Did you test it locally ?

**Comment by mamakancha** [comment]
Yes I ran the local tests using `./bin/test`

**Comment by mamakancha** [comment]
@bjodah Kindly review.

**Comment by bjodah** [comment]
This code doesn't actually work. I was surprised to see tests pass. Turns out we are testing `jn_zeros` with `method='scipy'` only using an older version of SciPy. You need to upgrade your local version of SciPy to see the failure:
https://asciinema.org/a/dm06ty5b4o82zp96o0jtgz6sk

**Comment by asmeurer** [comment]
The Travis tests should be fixed. We are installing scipy from apt, which is probably too old. We have a build that uses conda, but it is only testing the plotting tests.

**Comment by mamakancha** [comment]
@bjodah I have updated my scipy version
Also I ran, 
```
In [11]: type(spherical_jn(2,3.4))
Out[11]: numpy.float64

In [12]: type(spherical_jn(2,3.4)[0][-1])
---------------------------------------------------------------------------
IndexError                                Traceback (most recent call last)
<ipython-input-12-cf4711dd6e64> in <module>()
----> 1 type(spherical_jn(2,3.4)[0][-1])

IndexError: invalid index to scalar variable.
```

Can you suggest me hat should be done to get rid of `IndexError` in `Line 959`?

**Comment by bjodah** [comment]
@mamakancha yes, I believe that is the proper fix, you may look at the documentation to reassure yourself:
https://docs.scipy.org/doc/scipy-0.17.0/reference/generated/scipy.special.sph_jn.html
https://docs.scipy.org/doc/scipy/reference/generated/scipy.special.spherical_jn.html

See if you can update `.travis.yml` and/or `bin/test_travis.sh` as well so that we also test a later version of scipy using conda (as Aaron suggested).

**Comment by mamakancha** [comment]
@bjodah I have corrected the code .
Regarding the update of `.travis.yml` ,I am not quite acquainted with `travis`. Could you let me know what should be done ?

**Comment by mamakancha** [comment]
@bjodah  Can you please restart the falling **Theano** test?

**Comment by mamakancha** [comment]
ping @bjodah @jksuom

File: sympy/functions/special/bessel.py [hunk 1]
```diff
@@ -953,9 +953,13 @@ def jn_zeros(n, k, method="sympy", dps=15):
                                               int(l)), prec)
                 for l in range(1, k + 1)]
     elif method == "scipy":
-        from scipy.special import sph_jn
         from scipy.optimize import newton
-        f = lambda x: sph_jn(n, x)[0][-1]
+        try:
+            from scipy.special import spherical_jn
+            f = lambda x: spherical_jn(n, x)
+        except:
```
Comment (bjodah) [inline sympy/functions/special/bessel.py]: we could be explicit here and write `except ImportError:`

**Comment by bjodah** [comment]
Except from my (minor) comment I think it looks good!

File: sympy/functions/special/bessel.py (hunk 1 above)
Comment (mamakancha) [inline sympy/functions/special/bessel.py]: Thanks! Done :)
However, my doubt is , why do we use only `ImportError` and not except in general? To be aware of any kind of new error that may arise in future?

**Review by bjodah — APPROVED** [review]

**Comment by mamakancha** [comment]
@bjodah I hope this could be merged now?

**Comment by bjodah** [comment]
I just remembered that Aaron wanted to see Travis tests for this. I suppose it could be addressed in a separate PR.

**Comment by mamakancha** [comment]
Yes I would start working on it on a different PR with a little guidance by you if possible.

**Comment by bjodah** [comment]
Alright, @mamakancha, I've merged this. I'll make a note in the issue that we need tests for this still. Feel free to ping me from the new PR if you get stuck. You'll need to start looking at `.travis.yml` and figure out where the SciPy tests are run, it will most probably take you a while to figure it out (it would for me at least).

**Comment by mamakancha** [comment]
Thanks for the help. I will try to figure out what needs to be done :)

**Pull request merged**